config/dynamic.c

```diff
@@ -291,6 +292,175 @@ cleanup_dy(struct dynamic_config *dy_conf)
 		rte_free(dy_conf->server_path);
 		dy_conf->server_path = NULL;
 	}
+
+	destroy_mailbox(&dy_conf->mb);
+}
+
+static void
+process_return_message(lua_State *l, struct dynamic_config *dy_conf,
+	int num_succ_sent_inst)
+{
+	int num_gt_messages = 0;
+	size_t reply_len = 0;
+	char reply_msg[MSG_MAX_LEN];
+	int num_cmd_acc = 0;
+
+	/* Wait for all GT instances to synchronize. */
+	while (rte_atomic16_read(&dy_conf->num_returned_instances)
+			< num_succ_sent_inst)
+		rte_pause();
+
+	while (num_cmd_acc < num_succ_sent_inst) {
+		int i;
+		struct dy_cmd_entry *dy_cmds[dy_conf->mailbox_burst_size];
+		/* Load a set of commands from its mailbox ring. */
+		int num_cmd = mb_dequeue_burst(&dy_conf->mb,
+			(void **)dy_cmds, dy_conf->mailbox_burst_size);
```

Comment:
We need to break the loop here when `num_cmd == 0` to deal with the possibility that the GT blocks incremented `dy_conf->num_returned_instances` without sending a message due to not having enough memory to send the message. This condition is not trivial, so also add a commend explaining this reasoning.